apidef/graphql_config_adapter.go

```diff
@@ -0,0 +1,284 @@
+package apidef
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astparser"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/graphql_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/httpclient"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/rest_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/plan"
+	"github.com/jensneuse/graphql-go-tools/pkg/graphql"
+)
+
+var ErrUnsupportedGraphQLConfigVersion = errors.New("provided version of GraphQL config is not supported for this operation")
+
+type GraphQLConfigAdapter struct {
```

Comment:
Shouldn't this struct and all of its methods inside of the graphql library? I could not see gateway specific thing inside it.

Reply:
As an adapter, it should live where it adapts things - as it takes the apidef config and adapts it to the engine config the right place is the apidef package

Reply:
Then maybe, it is more than an adapter because I feel like `engineConfigV2DataSources` is not related to `apidef`